src/hotspot/cpu/aarch64/assembler_aarch64.hpp

```diff
@@ -221,22 +221,22 @@ class Instruction_aarch64 {
     mask <<= lsb;
     unsigned target = *(unsigned *)a;
     target &= ~mask;
-    target |= val;
+    target |= checked_cast<unsigned>(val);
```

Comment:
Any value that doesn't fit in 32 bits is going to fail, so it's tempting to force the callers to pass 32-bit types, but that's a bigger change.  How about something like this:

static ALWAYSINLINE void patch(address a, int msb, int lsb, uint32_t val) {
/* original code, no additional checked_cast needed */
}

static ALWAYSINLINE void patch(address a, int msb, int lsb, uint64_t val) {
  patch(a, msb, lsb, checked_cast<unsigned>(val));
}

Reply:
```
src/hotspot/cpu/aarch64/nativeInst_aarch64.hpp:614:35: error: call to 'patch' is ambiguous
[2023-07-11T14:00:01,568Z]   void set_kind(int order_kind) { Instruction_aarch64::patch(addr_at(0), 11, 8, order_kind); }
[2023-07-11T14:00:01,568Z]                                   ^~~~~~~~~~~~~~~~~~~~~~~~~~
```
The compiler thinks its ambiguous.

Reply:
OK, nevermind.